src/main/kotlin/no/nav/tiltakspenger/datadeling/vedtak/datadeling/VedtakService.kt

```diff
@@ -22,50 +30,96 @@ class VedtakService(
 
     /**
      * Merk at denne er reservert Arena og de ønsker at vi kun sender perioder bruker har rett til tiltakspenger.
-     * Ved en perfekt stans, ønsker de en tom liste.
+     * Hvis ingen gjeldende vedtaksperioder gir rett lenger, ønsker de en tom liste.
      * Ref: https://nav-it.slack.com/archives/CC9GYTA2C/p1734512113726549
      */
     fun hentTpVedtak(
         fnr: Fnr,
         periode: Periode,
     ): List<TiltakspengerVedtak> {
-        val toTidslinje = vedtakRepo.hentForFnrOgPeriode(fnr, periode, Kilde.TPSAK)
-            .filter { it.rettighet != TiltakspengerVedtak.Rettighet.AVSLAG }
-            .toTidslinje()
-        return toTidslinje
-            .filter { it.verdi.rettighet != TiltakspengerVedtak.Rettighet.STANS }
-            .map { it.verdi.oppdaterPeriode(it.periode) }
+        return hentInnvilgetTidslinje(fnr, periode)
+            .map { it.verdi.krympVirkningsperiode(it.periode) }
             .verdier
     }
 
     fun hentTidslinjeOgAlleVedtak(
         fnr: Fnr,
         periode: Periode,
     ): VedtakTidslinjeResponse {
-        val alleVedtak = vedtakRepo.hentForFnrOgPeriode(fnr, periode, Kilde.TPSAK)
-        val tidslinje = alleVedtak
-            .filter { it.rettighet != TiltakspengerVedtak.Rettighet.AVSLAG }
-            .toTidslinje()
-            .map { it.verdi.oppdaterPeriode(it.periode) }
+        val tidslinje = hentTidslinje(fnr, periode)
+            // Vil kunne inneholde både innvilgelser (inkl. omgjøringer) og stans.
+            .map { it.verdi.krympVirkningsperiode(it.periode) }
             .verdier
 
         return VedtakTidslinjeResponse(
             tidslinje = tidslinje.toVedtakResponse(logger),
-            alleVedtak = alleVedtak.toVedtakResponse(logger),
+            alleVedtak = vedtakRepo.hentForFnrOgPeriode(fnr, periode, Kilde.TPSAK).toVedtakResponse(logger),
         )
     }
 
+    /**
+     * Henter alle vedtak som påvirker rett til tiltakspenger i perioden fra både TPSAK og Arena.
+     * Avslag fra tp-sak ekskluderes.
+     */
     suspend fun hentVedtaksperioder(
         fnr: Fnr,
         periode: Periode,
     ): List<VedtakDTO> {
         val vedtakFraTpsak = vedtakRepo.hentForFnrOgPeriode(fnr, periode, Kilde.TPSAK)
-            .filter { it.rettighet != TiltakspengerVedtak.Rettighet.AVSLAG }
+            .filter { it.rettighet != AVSLAG }
+            // TODO jah: Ta en prat med Tia om dette er en OK løsning. Dette kan være et problem dersom klientene våre har lagret vedtakId'er fra omgjorte vedtak.
```

Comment:
Ser at denne brukes av modia-personoversikt, salesforce (usikker på om de faktisk bruker den) og tilleggsstønader. De to første lagrer ingenting, men jeg vet ikke hva tilleggstønader gjør. Foreslår at du hører med dem :)